gradle/publish-bintray.gradle

```diff
@@ -12,72 +12,149 @@ apply plugin: "com.github.johnrengelman.shadow"
+                it.artifactId.each { node ->
+                    def artifactId = node.text()
+                    if (!artifactId.endsWith("native")) return
+
+                    switch (project.name) {
+                        case 'metadata':
+                            node.setValue("${artifactId[0..-8]}-common")
+                            break
+                        case 'js':
+                            node.setValue("${artifactId[0..-8]}-js")
+                            break
+                    }
+                }
+            }
+        }
     }
 }
 
+def isMultiplatform = project.name == "kotlinx-coroutines-core"
 
 publishing {
     repositories {
         maven { url = 'https://kotlin.bintray.com/kotlinx' }
     }
 
-    publications {
-        maven(MavenPublication) { publication ->
-            preparePublication(publication)
+    if (!isMultiplatform) {
+        publications {
+            maven(MavenPublication) { publication ->
+                if (project.name == "kotlinx-coroutines-debug") {
+                    project.shadow.component(publication)
+                } else {
+                    publication.from components.java
+                }
+                publication.artifact javadocJar
+                publication.artifact sourcesJar
+                publication.pom.withXml(configureMavenCentralMetadata)
+            }
         }
-    }
-}
 
-artifactory {
-    contextUrl = 'https://oss.jfrog.org/artifactory'
-    publish {
-        repository {
-            repoKey = 'oss-snapshot-local'
-            username = bUser
-            password = bKey
-        }
+        return
+    }
 
-        maven(MavenPublication) { publication ->
-            preparePublication(publication)
+    // Rename artifacts for backward compatibility
+    publications.all {
+        def type = it.name
+        switch (type) {
+            case 'kotlinMultiplatform':
+                it.artifactId = "$project.name-native"
+                it.artifact emptyJar
+                it.artifact stubJavadoc
+                it.artifact sourcesJar
+                break
+
+            case 'metadata':
+                it.artifactId = "$project.name-common"
+                break
+
+            case 'jvm':
+                it.artifactId = "$project.name"
+                break
+            case 'js':
+            case 'native':
+                it.artifactId = "$project.name-$type"
+                break
         }
 
-        defaults {
-            publications('maven')
-        }
+        pom.withXml(configureMavenCentralMetadata)
+        configureTransitiveDependencies(project, it)
     }
+
+    disableMetadataPublication()
 }
 
-def preparePublication(MavenPublication publication) {
-    if (!isNative()) {
-        if (project.name == "kotlinx-coroutines-debug") {
-            project.shadow.component(publication)
-        } else {
-            publication.from components.java
+private void disableMetadataPublication() {
```

Comment:
Something is wrong with the way this code is written. Metadata publishing is (correctly) disabled for `kotlinx-coroutines`, `kotlinx-coroutines-common`, and `kotlinx-coroutines-js`, but there is metadata for all additional (non-multiplatform) gradle subprojects (`kotlinx-coroutines-jdk8`, `kotlinx-coroutines-guava`, etc)